file_cache_test.go

```diff
@@ -752,11 +762,88 @@ var _ = Describe("FileCache", func() {
 							"/diego.txt",
 							"/testdir",
 							"/testdir/file.txt",
+							"/testdir/fileLink.txt",
 						))
 					})
 				})
 			})
 
+			Context("and the directory is closed", func() {
+				JustBeforeEach(func() {
+					dir, cacheInfoType, getErr = cache.GetDirectory(cacheKey)
+					Expect(getErr).NotTo(HaveOccurred())
+
+					err := cache.CloseDirectory(cacheKey, dir)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				Context("and the directory is later fetched", func() {
+					JustBeforeEach(func() {
+						dir, cacheInfoType, getErr = cache.GetDirectory(cacheKey)
+						Expect(getErr).NotTo(HaveOccurred())
+						Expect(dir).NotTo(BeEmpty())
+						Expect(cacheInfoType).To(Equal(cacheInfo))
+					})
+
+					It("returns a valid path", func() {
```

Comment:
Sorry yeah I merged some code in and there are a few tests that got added again accidentily.  Did you want to just fix them or want me to update and resubmit?